src/Spice86.Core/Emulator/InterruptHandlers/VGA/VgaFunctionality.cs

```diff
@@ -17,28 +17,33 @@ public class VgaFunctionality : IVgaFunctionality {
     private readonly BiosDataArea _biosDataArea;
     private readonly IIOPortHandler _ioPortDispatcher;
     private readonly IIndexable _memory;
-    private readonly VgaRom _vgaRom;
     private readonly InterruptVectorTable _interruptVectorTable;
 
     /// <summary>
     /// Creates a new instance of the <see cref="VgaFunctionality"/> class.
     /// </summary>
+    /// <param name="interruptVectorTable">The interrupt vector table wrapper.</param>
     /// <param name="memory">The emulator memory.</param>
     /// <param name="ioPortDispatcher">The IOPortDispatcher, used to read from or write to VGA ports.</param>
     /// <param name="biosDataArea">The global BIOS variables.</param>
-    /// <param name="vgaRom">The VGA ROM, so we can access the IBM fonts.</param>
     /// <param name="bootUpInTextMode">Whether we begin with mode 0x03.</param>
-    public VgaFunctionality(IIndexable memory, IIOPortHandler ioPortDispatcher, BiosDataArea biosDataArea, VgaRom vgaRom, bool bootUpInTextMode) {
+    public VgaFunctionality(InterruptVectorTable interruptVectorTable, IMemory memory, IIOPortHandler ioPortDispatcher, BiosDataArea biosDataArea, bool bootUpInTextMode) {
         _memory = memory;
         _ioPortDispatcher = ioPortDispatcher;
         _biosDataArea = biosDataArea;
-        _vgaRom = vgaRom;
-        _interruptVectorTable = new (memory);
+        _interruptVectorTable = interruptVectorTable;
+        VgaRom = new VgaRom();
```

Comment:
This kinda makes it impossible to use a different VGA rom.

But if you really want to make the rom part of the functionality (rather than the functionality depending on the rom) they you can remove it from the Machine too and make the VgaRom property private. Nothing else depends on the VgaRom.

Reply:
- My plan for the entire video emu code was to revert all my changes, this should have no changes compared to Master '

Reply:
Fixed.